src/utils.ts

```diff
@@ -59,6 +61,15 @@ export const ByteUtils = {
   }
 };
 
+/**
+ * Use this to test if an object is a string. This is because
+ * typeof new String('test') is 'object' and not 'string'.
```

Comment:
Just curious, why do you want to accept `new String('test')` here?

Reply:
Just defensive if a user decided to define their strings in that manner.

Reply:
It’s your call ultimately but I think it’s highly unlikely that users would do this (or at least would do this intentionally).

Reply:
Per conversation with Neal last week I'll change this to "typeof" since we don't generally handle `new String()` in the driver or BSON as a string, but rather an object.